File: packages/emitter/testcases/arrays/methods/ArrayMethods.ts
export function processArray(arr: number[]): number {
  const doubled = arr.map((x: number): number => x * 2);
  const filtered = doubled.filter((x: number): boolean => x > 5);
  const sum = filtered.reduce((acc: number, x: number): number => acc + x, 0);
  return sum;
}

File: packages/emitter/testcases/js-only/arrays/methods/ArrayMethods.cs
namespace TestCases.jsonly.arrays.methods
{
        public static class ArrayMethods
        {
            public static double processArray(global::System.Collections.Generic.List<double> arr)
                {
                var doubled = global::Tsonic.JSRuntime.Array.map(arr, (double x) => x * 2);
                var filtered = global::Tsonic.JSRuntime.Array.filter(doubled, (double x) => x > 5);
                var sum = global::Tsonic.JSRuntime.Array.reduce(filtered, (double acc, double x) => acc + x, 0);
                return sum;
                }
        }
}